vscode/webviews/chat/cells/messageCell/human/HumanMessageCell.tsx

```diff
@@ -96,34 +97,58 @@ const HumanMessageCellContent = memo<HumanMessageCellContent>(props => {
         intent,
     } = props
 
+    const containerRef = useRef<HTMLDivElement>(null)
+
+    useEffect(() => {
+        if (!containerRef.current) return
+        const handleCopy = (event: ClipboardEvent) => {
+            const selectedText = window.getSelection()?.toString() || ''
+            getVSCodeAPI().postMessage({
+                command: 'copy',
+                text: selectedText,
+                eventType: 'Keydown',
+            })
+        }
+
+        containerRef.current.addEventListener('copy', handleCopy)
+
+        return () => {
+            if (containerRef.current) {
+                containerRef.current.removeEventListener('copy', handleCopy)
+            }
+        }
+    }, [])
```

Comment:
```suggestion
    // Setup global copy event listener for chat view
    useEffect(() => {
        const handleCopyEvent = async (event: ClipboardEvent) => {
            const target = event.target
            // Check if the target or target's parent is a code element
            if (target instanceof HTMLElement && target.closest('code')) {
                // Not sure how to check if this is a keydown or mouse event :D 
                navigator.clipboard.readText().then(text => copyButtonOnSubmit(text, 'Keydown'))
            }
        }

        document.addEventListener('copy', handleCopyEvent)
        return () => {
            document.removeEventListener('copy', handleCopyEvent)
        }
    }, [copyButtonOnSubmit])
```

I don't think we will need the container ref if we check if the copied element is inside a code block or not? We can move this to Transcript (?) that way we don't need to set up a listener for each code block? wdyt?